CONTRIBUTING.md

```diff
@@ -85,6 +89,75 @@ Because we will only merge a Pull Request for which all tests pass. The followin
   See [About TypeScript demos](#about-typescript-demos).
 - The Pull Request title follows the pattern `[Component] Imperative commit message`. (See: [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#imperative) for a great explanation)
 
+#### Checks and how to fix them
+
+If any of the checks fails click on the _Details_
+link and checkout the logs of the build to find out why it failed. The following
+section gives an overview what each check is responsible for.
+
+##### ci/codesandbox
+
+This task should not fail in isolation. It creates codesandboxes that use the version
+of Material-UI that was built from this Pull Request. Use it to test more complex scenarios.
+
+##### ci/circleci: checkout
+
+A preflight check to see if the dependencies and lockfile is ok. Running `yarn`
+and `yarn deduplicate` should fix most of the issues.
+
+##### ci/circleci: test_static
+
+Checks code format and lints the repository. The log of the failed build should explain
+how to fix the issues.
+
+##### ci/circleci: test_unit-1
+
+Runs the unit tests in a `jsdom` environment. If this fails then `yarn test:unit`
+should fail locally as well.
```

Comment:
```suggestion
should fail locally as well. (You can narrow the scope of tests run with `yarn test:unit -g ComponentName`.)
```
?

Reply:
Yes this is definitely important. I'll add a link to the `--grep` documentation. Nowadays I use e.g. `"<Tab />"` to avoid also running tests for Tabs. Maybe there are some hints how to grep for test filename.

Reply:
Should we recommend `yarn test:watch` for faster repeated runs?

Reply:
Not right now. It's broken.